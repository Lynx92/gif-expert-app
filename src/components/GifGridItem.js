import React from "react";
import PropTypes from "prop-types";

function GifGridItem({ title, url }) {
  return (
    <div className="card animate__animated animate__bounceInUp">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
