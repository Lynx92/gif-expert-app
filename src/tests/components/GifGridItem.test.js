import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("GifGridItem", () => {
  const title = "Titulo";
  const url = "http://localhost/algo.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("has p with title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("has an img with url and alt props", () => {
    const img = wrapper.find("img");
    const { src, alt } = img.props();
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("has correct classes", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("card")).toBe(true);
    expect(div.hasClass("animate__animated")).toBe(true);
    expect(div.hasClass("animate__bounceInUp")).toBe(true);
  });
});
