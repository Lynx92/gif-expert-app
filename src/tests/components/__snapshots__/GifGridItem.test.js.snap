// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem renders correctly 1`] = `
<div
  className="card animate__animated animate__bounceInUp"
>
  <img
    alt="Titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
